fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ExecuteCommand.java

```diff
@@ -70,14 +70,15 @@ public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
         LogicalPlanAdapter planAdapter = new LogicalPlanAdapter(prepareCommand.getLogicalPlan(), executor.getContext()
                 .getStatementContext());
         executor.setParsedStmt(planAdapter);
-        // If it's not a short circuit query or schema version is different(indicates schema changed),
-        // need to do reanalyze and plan
+        // If it's not a short circuit query or schema version is different(indicates schema changed) or
+        // has nondeterministic functions in statement, then need to do reanalyze and plan
         boolean isShortCircuit = executor.getContext().getStatementContext().isShortCircuitQuery();
         boolean hasShortCircuitContext = preparedStmtCtx.shortCircuitQueryContext.isPresent();
         boolean schemaVersionMismatch = hasShortCircuitContext
                     && preparedStmtCtx.shortCircuitQueryContext.get().tbl.getBaseSchemaVersion()
                     != preparedStmtCtx.shortCircuitQueryContext.get().schemaVersion;
-        boolean needAnalyze = !isShortCircuit || schemaVersionMismatch || !hasShortCircuitContext;
+        boolean needAnalyze = !isShortCircuit || schemaVersionMismatch || !hasShortCircuitContext
+                        || executor.getContext().getStatementContext().hasNondeterministicInPreparedStatement();
```

Comment:
does it work? ExpressionAnalyzer be called in nerieds' analyzer. ExpressionAnalyzer should not be called when come here. so i don't think hasNondeterministicInPreparedStatement will be set correctly.

Reply:
i got it. u save the statement context and reuse them between these execute command. it works